src/main/java/jp/ikedam/jenkins/plugins/extensible_choice_parameter/SystemGroovyChoiceListProvider.java

```diff
@@ -328,4 +333,33 @@ protected void setProject(AbstractProject<?,?> project)
     {
         return project;
     }
+    
+    public static class ConverterImpl extends XStream2.PassthruConverter<SystemGroovyChoiceListProvider> {
+        private final Mapper mapper;
+        
+        public ConverterImpl(XStream2 xstream) {
+            super(xstream);
+            mapper = xstream.getMapper();
+        }
+        
+        @Override
+        public void marshal(Object source, HierarchicalStreamWriter writer, MarshallingContext context) {
+            super.marshal(source, writer, context);
+            
+            // As project is transient, serialize it forcibly.

```

Comment:
This is generally a bad idea. Rather assume the field will be null after deserialization and set it again from some enclosing context, such as an `onLoad` method; or (better) get rid of the field altogether, and pass the project as a parameter to methods which actually require it.